a10_octavia/controller/worker/tasks/server_tasks.py

```diff
@@ -222,3 +238,66 @@ def execute(self, member, vthunder, pool, pool_count_ip, member_count_ip_port_pr
         except exceptions.ConnectionError as e:
             LOG.exception("Failed to delete member/port: %s", member.id)
             raise e
+
+
+class MemberFindNatPool(MemberBaseTask):
+
+    @axapi_client_decorator
+    def execute(self, member, vthunder, pool, flavor=None):
+        if flavor is None:
+            return
+
+        pool_flavor = flavor.get('nat_pool')
+        pools_flavor = flavor.get('nat_pool_list')
+        if pool_flavor or pools_flavor:
+            for listener in pool.listeners:
+                vport = self.axapi_client.slb.virtual_server.vport.get(pool.load_balancer.id,
+                                                                       listener.id,
+                                                                       listener.protocol,
+                                                                       listener.protocol_port)
+                if vport and 'port' in vport and 'pool' in vport['port']:
+                    if pool_flavor and vport['port']['pool'] == pool_flavor['pool_name']:
+                        return pool_flavor
+                    for flavor in (pools_flavor or []):
+                        if vport['port']['pool'] == flavor['pool_name']:
+                            return flavor
+
+
+class MemberReserveSubnetAddr(MemberBaseTask):
+
+    def execute(self, member, nat_flavor=None, nat_pool=None):
+        if nat_flavor is None:
+            return
+
+        if nat_pool is None:
+            try:
+                addr_list = []
+                start = (struct.unpack(">L", socket.inet_aton(nat_flavor['start_address'])))[0]
+                end = (struct.unpack(">L", socket.inet_aton(nat_flavor['end_address'])))[0]
+                while start <= end:
+                    addr_list.append(socket.inet_ntoa(struct.pack(">L", start)))
+                    start += 1
+                port = self.network_driver.reserve_subnet_addresses(member.subnet_id, addr_list)
+                LOG.debug("Successfully allocate addresses for nat pool %s on port %s",
+                          nat_flavor['pool_name'], port.id)
+                return port
+            except Exception as e:
+                LOG.exception("Failed to reserver addresses in NAT pool %s from subnet %s",
```

Comment:
typo : `  LOG.exception("Failed to reserve addresses in NAT pool %s from subnet %s",`

Reply:
ok